enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,179 @@
+---
+title: enable-baremetal-on-other-platforms
+authors:
+  - "@asalkeld"
+  - "@sadasu"
+reviewers:
+  - TBD
+  - "@hardy"
+  - "@romfreiman"
+  - "@dhellman"
+approvers:
+  - TBD
+creation-date: 2021-08-20
+last-updated: 2021-08-20
+status: implementable
+see-also:
+  - "/enhancements/baremetal-provisiong-config.md"
+replaces:
+superseded-by:
+---
+
+# Enable Baremetal on other Platforms to support ZTP
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+BMO is only available when deploying an OpenShift cluster with the baremetal
+platform (via the IPI or AI workflow). Having the ability to manage baremetal nodes from
+clusters without requiring the cluster to be on baremetal would be beneficial
+to customers.
+
+## Motivation
+
+An initial driver of this feature are the centralized host management use cases
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on bare metal.
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the ZTP use case by partially enabling cluster-baremetal-operator on the following platforms:
+- None
+- OpenStack
+- vSphere
+
+We will be successful when:
+
+ZTP can deploy clusters using the above platforms.
+
+### Non-Goals
+
+Allow CBO to be fully enabled on all platforms.
+
+Note that the scope of this epic is only enablement of the Metal3
+BaremetalHost API and associated controller/services - the edge ZTP flow
+currently interacts directly via this API without any Machine API integration.
+
+## Proposal
+
+Currently cluster-baremetal-operator checks the platform and if it is not
+baremetal it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow cluster-baremetal-operator to be enabled for the
+ZTP use case which is Platforms: None, OpenStack or vSphere.
+
+Further (to restrict the testing matrix) the allowed configuration options
+of the Provisioning CR will be restricted to exactly those required by ZTP.
+
+*Only spec.provisioningNetwork=Disabled mode will be accepted in the Provisioning CR.*
+
+Note: Since BMO (and metal3) is only required to manage existing nodes in this
+use case, there is no requirement to scale the Baremetal machineSet. So no integration
+with the Machine API is required.
+
+### User Stories
+
+#### Story 1 - Current IPI baremetal platform use case
+
+No change.
+
+#### Story 2 - ZTP use case
+
+If the platform is one of "None, OpenStack or vSphere". cluster-baremetal-operator
```

Comment:
This text describes how to implement a solution, not a user story. This text would be good to re-locate into the "Proposal" section. The user story section should focus on what a user needs to accomplish.

A user story would be: "As a user of a hub cluster that performs central infrastructure management, and optionally zero-touch provisioning, I need to provision hosts using a k8s-native API even when the hub cluster has a platform of None, OpenStack, or vSphere."